sklearn/inspection/_plot/tests/test_boundary_decision_display.py

```diff
@@ -577,16 +575,81 @@ def test_class_of_interest_multiclass(pyplot, response_method):
             class_of_interest=class_of_interest_idx,
         )
 
-    # TODO: remove this test when we handle multiclass with class_of_interest=None
-    # by showing the max of the decision function or the max of the predicted
-    # probabilities.
-    err_msg = "Multiclass classifiers are only supported"
-    with pytest.raises(ValueError, match=err_msg):
+
+@pytest.mark.parametrize("response_method", ["predict_proba", "decision_function"])
+def test_multiclass_plot_max_class(pyplot, response_method):
+    """Check plot correct when plotting max multiclass class."""
+    X, y = load_iris_2d_scaled()
+    clf = LogisticRegression().fit(X, y)
+
+    disp = DecisionBoundaryDisplay.from_estimator(
+        clf,
+        X,
+        response_method=response_method,
+    )
+
+    grid = np.concatenate([disp.xx0.reshape(-1, 1), disp.xx1.reshape(-1, 1)], axis=1)
+    response = getattr(clf, response_method)(grid)
+    assert_allclose(response.reshape(*disp.response.shape), disp.response)
+
+
+@pytest.mark.parametrize(
+    "multiclass_colors",
+    [
+        "plasma",
+        ["red", "green", "blue"],
+    ],
+)
+@pytest.mark.parametrize("plot_method", ["contourf", "contour", "pcolormesh"])
+def test_multiclass_colors_cmap(pyplot, plot_method, multiclass_colors):
+    """Check correct cmap used for all `multiclass_colors` inputs."""
+    import matplotlib as mpl
+
+    X, y = load_iris_2d_scaled()
+    clf = LogisticRegression().fit(X, y)
+
+    disp = DecisionBoundaryDisplay.from_estimator(
+        clf,
+        X,
+        plot_method=plot_method,
+        multiclass_colors=multiclass_colors,
+    )
+
+    cmaps = []
+    if multiclass_colors == "plasma":
+        colors = mpl.pyplot.get_cmap(multiclass_colors, len(clf.classes_)).colors
+    else:
+        colors = [mpl.colors.to_rgba(color) for color in multiclass_colors]
+
+    for class_idx, (r, g, b, _) in enumerate(colors):
+        class_cmap = mpl.colors.LinearSegmentedColormap.from_list(
+            f"colormap_{class_idx}", [(1.0, 1.0, 1.0, 1.0), (r, g, b, 1.0)]
+        )
+        cmaps.append(class_cmap)
+
+    for idx, quad in enumerate(disp.surface_):
+        # Ensure `_lut` (look-up table) is equal
+        quad.cmap._init()
+        cmaps[idx]._init()
+        assert_array_equal(quad.cmap._lut, cmaps[idx]._lut)
```

Comment:
Okay I think I will use the `==` for mpl versions >=3.5 and check for bounds for lower versions. I will add a TODO to remove the latter check once our min mpl version is 3.5. I think this would be ideal, as `==` is what we ideally want to use.